File: src/JsonServices.CSharp/Messages/RequestMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JsonServices.Messages
{
	public class RequestMessage<T> : IRequestMessage
	{
		public RequestMessage()
		{
			Name = typeof(T).FullName;
		}

		public string Id { get; set; }

		public bool IsOneWay => string.IsNullOrWhiteSpace(Id);

		public string Name { get; set; }

		public T Params { get; set; }

		object IRequestMessage.Params => Params;
	}
}

File: sample/sample-client/src/JsonServices/RequestMessage.ts
export class RequestMessage {
    constructor(method: string, params: object, id?: string) {
        this.method = method;
        this.params = params;
        this.id = id;
    }

    public jsonrpc = "2.0";
    public method: string;
    public params: object;
    public id?: string;
}